Automattic/wp-calypso
# Component: add additional feedback form to the NPS survey

This PR will add an additional form to the NPS survey dialog to get contextual feedback. The task is a part of the NPS v2 project. To see more details of the project, please visit p9jf6J-h6-p2

## How to test

1. Run `localStorage.setItem('debug', 'calypso:nps-survey');` in the browser console to track the logs.
1. Execute `npsSurvey();` as well to see the NPS notification at the Reader page.
1. In one or more seconds, the notification should show up as follows:
    <img width="721" alt="2018-03-26 10 56 34" src="https://user-images.githubusercontent.com/212034/37911249-83b178f6-314a-11e8-8d88-748b481d797c.png">
1. Click on the "Sure!" button and you will see the NPS survey form.
    <img width="816" alt="following_ _reader_ _wordpress_com" src="https://user-images.githubusercontent.com/212034/37911277-94cce814-314a-11e8-8c1d-232158a08d22.png">
1. When you click on the **Submit** button, the contextual feedback textbox appears.
    <img width="816" alt="following_ _reader_ _wordpress_com-2" src="https://user-images.githubusercontent.com/212034/37911370-d01a2d78-314a-11e8-9a69-5db985d141d9.png">
1. Input text, submit the form and the console shows you `Sending NPS survey feedback...`.
1. Depending on the result, the console shows `...Successfully sent NPS survey feedback` or `...Error sending NPS survey feedback`.
